2380 document featsecret introduce secret management (#2431)
risingwavelabs/risingwave-docs: docs/deploy/manage-secrets.md

description: Use the secret management feature to store credentials securely.
slug: /manage-secrets
---
<head>
  <link rel="canonical" href="https://docs.risingwave.com/docs/current/manage-secrets/" />
</head>

:::note Premium Edition Feature
This feature is only available in the premium edition of RisingWave. The premium edition offers additional advanced features and capabilities beyond the free and community editions. If you have any questions about upgrading to the premium edition, please contact our sales team.
:::

By default, credentials for connecting to external services (like MySQL) are specified in plain text within the `WITH` clause of `CREATE SOURCE / SINK` statements. This practice poses security risks, particularly for large organizations where multiple teams manage connected services.

To address the issue, we recommend using the `CREATE SECRET` command to store credentials securely. Admins can create secrets in advance, allowing other team members to reference them using secret identifiers when creating source/sink connections. This ensures that secrets remain protected throughout all phases of access.

RisingWave provides four key secret management operations:

> - Creating secrets.
> 
> - Using secrets.
> 
> - Using secrets as a file.
> 
> - Dropping secrets.

Reviewer: ```suggestion
- Create secrets.

- Use secrets.

- Use secrets as a file.

- Drop secrets.
```

Author: thanks, this is not a heading. So I use ing.

In addition, you can use the [`rw_secrets`](/sql/system-catalogs/rw_catalog.md) catalog to view the ID, name, owner, and access control of secret objects.

## Create secrets

You can use the following statement to create secrets:

```sql title="Syntax for creating secrets"
CREATE SECRET secret_name WITH ( backend = 'meta') AS 'your_secret';
```

```sql title="Examples"
CREATE SECRET mysql_pwd WITH (
  backend = 'meta'
) AS '123';
```

:::note
Currently only the meta backend is supported.
:::

> ## Use Secrets

Reviewer: ```suggestion
## Use secrets
```

After creating secrets, you can use `SECRET your_secret_name` as the option value in the `WITH` clause. For example:

```sql title="Use a secret in the WITH clause"
CREATE SECRET mysql_pwd WITH (
  backend = 'meta'
) AS '123';

CREATE SOURCE mysql_source WITH (
connector = 'mysql-cdc',
hostname = 'localhost',
port = '8306',
username = 'rwcdc',
password = secret mysql_pwd,
database.name = 'test',
server.id = '5601'
);
```

## Use secrets as a file

Some connectors need credentials stored as file paths (e.g., `ssl.ca.location`), where the file contains the secret. RisingWave allows you to reference a secret as a file path.

```sql title="Reference a secret as a file path"
CREATE TABLE district (
    d_id INTEGER,

>     PRIMARY KEY (d_id, d_w_id)

Reviewer 2: ```suggestion
PRIMARY KEY (d_id)
```
